src/com/xilinx/rapidwright/timing/TimingModel.java

```diff
@@ -500,17 +491,93 @@ protected List<TimingGroup> determineGroups(List<Node> nodes, List<IntentCode> n
         return result;
     }
 
+    private Pair<Integer,Integer> findReferenceTileLocation(Device d) {
+        class Helper {
+            Map<Integer,Set<Integer>> spans;
+            public Helper() {
+                spans = new HashMap<Integer,Set<Integer>>();
+            }
+            void insert(int val, int span) {
+                if (!spans.containsKey(span)) {
+                    spans.put(span, new HashSet<Integer>());
+                }
+                spans.get(span).add(val);
+            }
+            Set<Integer> longest() {
+                List<Integer> keys = new ArrayList<>(spans.keySet());
+                Integer maxSpan = Collections.max(keys);
+                return spans.get(maxSpan);
+            }
+        }
+
+
+        // for each column, look for valid row
+        Helper colHelper = new Helper();
+        for (int i = 0; i < d.getColumns(); i++) {
+            int span = 0;
+            for (int j = 0; j < d.getRows(); j++) {
+                if (d.getTile("INT_X" + i + "Y" + j) != null) {
+                    span++;
+                }
+            }
+            colHelper.insert(i,span);
+        }
+
+        // for each row, look for valid col
+        Helper rowHelper = new Helper();
+        for (int j = 0; j < d.getRows(); j++) {
+            int span = 0;
+            for (int i = 0; i < d.getColumns(); i++) {
+                if (d.getTile("INT_X" + i + "Y" + j) != null) {
+                    span++;
+                }
+            }
+            rowHelper.insert(j,span);
+        }
+

```

Comment:
```suggestion

        // Find the tile with the most number of INT tiles on the same row/col
```

What's the significance of most number of INT tiles? Would some upper limit (e.g. equivalent to the longest wire) be sufficient?

Reply:
The model needs to compute "d" which is a function of tile types. Two long wires may span different set of tiles. Thus, need to scan the whole device to build distance array for both vertical and horizontal. Thus, one full row (column) is required.